src/logic/embedding.lean

```diff
@@ -320,3 +320,64 @@ namespace set
 ⟨λ x, ⟨x.1, h x.2⟩, λ ⟨x, hx⟩ ⟨y, hy⟩ h, by { congr, injection h }⟩
 
 end set
+
+section subtype
+
+variable {α : Type*}
+
+/-- A subtype `{x // p x ∨ q x}` over a disjunction of `p q : α → Prop` can be injectively split
+into a sum of subtypes `{x // p x} ⊕ {x // q x}` such that `¬ p x` is sent to the right. -/
+def subtype_or_embedding (p q : α → Prop) [decidable_pred p] :
```

Comment:
It might make sense to put `left` in the name, since you're giving the left half of the `or` priority in case of a tie.